src/System.CommandLine.Subsystems/CalculatedValue.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.CommandLine.ValueSources;
+
+namespace System.CommandLine;
+
+/// <summary>
+/// CalculatedValueSymbol lets ValueSource contribute to the data space of the application
```

Comment:
This doesn't match the class name. I would lean towards CalculatedValueSymbol?

Reply:
Per discussions with Mikayla, we want to explore using custom type providers in the core layer to solve the problems calculated values is solving in the subsystem layer in this PR. 

For clarity, I am closing this PR and replacing it with a PR that has the other work that was included in this PR.